Directory path of library install name removed

tanshihaj opened the pull request:
Currently library install name after packaging is something like `/Users/travis/.conan/data/libffi/3.2.1/bincrafters/stable/package/ea598ec51f506a4c0ce579fe1e9cbc6299458ebb/lib/libffi.6.dylib` and therefore it is impossible to link this library without applying `install_name_tool -id ...`. It seems that default conan approach is to remove directory part of library install name and leave only library name, so this patch removes directory part of install name in `configure` script.

tanshihaj committed:
Directory path of library install name removed

madebr commented:
I think you can move the `tools.replace_in_file` to the `source` function and apply it unconditionally.

uilianries committed:
Apply changes on build stage

Since the source folder should have the same contents for all platforms, I would prefer to patch all files when build is called.

uilianries commented:
Hi @tanshihaj !

Thanks for contributing! I just updated your PR, your PR is right but I moved all patches for the build stage.

@madebr I try to keep the source folder untouchable. When you have any problem and need to check the files, you can use the source folder as raw project, without any change. Also, as many platforms are using the same package, it will be nice have the exactly same folder for all platforms.
